src/dialogs/config-flow/step-flow-form.ts

```diff
@@ -136,7 +136,7 @@ class StepFlowForm extends LitElement {
     const toSendData = {};
     Object.keys(stepData).forEach((key) => {
       const value = stepData[key];
-      const isEmpty = [undefined, ""].includes(value);
+      const isEmpty = value === undefined;
```

Comment:
I think this should depend on the type of field, so the logic should probably be moved to the individual form elements, (maybe we already do, have not checked) for a number for instance, we never want to send an empty string...

Reply:
And not sure if the field is inited to an empty string when you start? If we send an empty string when the user has not entered anything, integrations might handle that wrong, a check for `None` will now fail? (Again not checked in the code) we should probably only send an empty string if the field was set already? Not sure...

Reply:
What about something like this:
```ts
      if (!isEmpty) {
        toSendData[field.name] = value;
      } else if (field.optional) {
        toSendData[field.name] = null;
      }
```

So we would send `null` (`None`), for every empty optional field...
This would be a major breaking change, as both voluptuous serialize and every config flow with optional fields will need to be updated to allow `None`
Or we would add support for it in voluptuous serialize, and depending on that we send `None` for empty optional fields... So a config flow could opt-in to this behavior

Reply:
The cases where we want to clear something that I know of are all lists of values (usually CSV).  I wonder if there is a multi-select control that can add/remove values that makes more sense to use here?

Reply:
There is multi select support in config flows

Reply:
Yes, free form values would be a much better UX vs asking users to enter CSV data inline.  It solves this issue since we would send an empty list to clear out the value.